### PR TITLE
adding matrix addition

### LINKED ISSUE #34: Matrix addition , multiplication and transpose
## 🚀 Feature

Hi, I am a GSSoC participant I would like to implement this in java, python ,c, c++, and javascript

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Pitch

implement matrix addition multiplication and transpose of a matrix

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Fixes #34 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #34

### CODE REVIEW

File: docs/mathematics/Matrix-Addition.mdx [hunk 1]
```diff
@@ -0,0 +1,177 @@
+---
+title: Matrix Addition
+slug: /mathematics/Matrix-Addition
```
Comment (HarshCasper) [inline docs/mathematics/Matrix-Addition.mdx:3]: Keep the slug lower-case

File: docs/mathematics/Matrix-Addition.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+---
+
+## Introduction
+
+In mathematics, matrix addition is the operation of adding two matrices by adding the corresponding entries together. Two matrices must have an equal number of rows and columns to be added.
+
+## Explanation
+
+Consider two matrices A and B , the sum of two matrices A and B will be a matrix which has the same number of rows and columns as A and B. The sum of A and B, denoted A + B, It is computed by adding corresponding elements of A and B
```
Comment (HarshCasper) [inline docs/mathematics/Matrix-Addition.mdx:12]: Can you make a diagram using some prototyping tool and add them here?

File: docs/mathematics/Matrix-Addition.mdx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+## Algorithm
+
+Input matrix A and matrix B.<br/>
+Input matrix 1 <br/>
+Input matrix 2 <br/>
+If the number of rows and number of columns of matrix A and matrix B is equal<br/>
+for i=0 to number of rows in A do<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;for j=0 to number of rows in A do<br/> 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        Result [i,j] =  A [i,j]+ B [i,j]<br/>
+Display Result
```
Comment (HarshCasper) [inline docs/mathematics/Matrix-Addition.mdx:24]: Keep in a bullet list. Use the algorithm just for stating the steps. Don'r use any code/pseudo-code here

File: docs/mathematics/Matrix-Addition.mdx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+13 9 10
+9 9 14
+```
+
+## Complexity Analysis
+
+```
+Time complexity - O(n * m)
+Space complexity - O(n*m),, where 'n' and 'm' are the number of rows and columns respectively
+```
```
Comment (HarshCasper) [inline docs/mathematics/Matrix-Addition.mdx:172]: Don't keep them in a triple backticks. Keep them as a bullet-point, and add backticks on the values and around `n` and `m`

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please fix the merge conflicts

File: docs/mathematics/Matrix-Addition.mdx (hunk 2 above)
Comment (aaadddiii) [reply docs/mathematics/Matrix-Addition.mdx:12]: I tried to insert an image using markdown syntax but it is not working here

Comment (HarshCasper) [reply docs/mathematics/Matrix-Addition.mdx:11]: Just use `image`

**Comment by HarshCasper** [comment]
@aaadddiii The CI has been broken. It might have been because of an error on the `sidebars.js` file

File: docs/mathematics/Matrix-Addition.mdx (hunk 2 above)
Comment (aaadddiii) [reply docs/mathematics/Matrix-Addition.mdx:11]: Sorry @HarshCasper It was an issue with the image url

**Comment by HarshCasper** [comment]
@aaadddiii Test it locally using `yarn start` and `yarn build` commands before pushing anything on the PR.

**Comment by aaadddiii** [comment]
@HarshCasper  It is working on my local server but it's failing here

**Comment by HarshCasper** [comment]
> @HarshCasper It is working on my local server but it's failing here

Can you do a `yarn build` on your local machine?

**Comment by aaadddiii** [comment]
> > @HarshCasper It is working on my local server but it's failing here
> 
> Can you do a `yarn build` on your local machine?

I have done `yarn build` 
image

**Comment by HarshCasper** [comment]
> > > @HarshCasper It is working on my local server but it's failing here
> > 
> > 
> > Can you do a `yarn build` on your local machine?
> 
> I have done `yarn build`
> image

Can you push again?

**Comment by aaadddiii** [comment]
> > > > @HarshCasper It is working on my local server but it's failing here
> > > 
> > > 
> > > Can you do a `yarn build` on your local machine?
> > 
> > 
> > I have done `yarn build`
> > image
> 
> Can you push again?

Yes

**Comment by aaadddiii** [comment]
> > > > @HarshCasper It is working on my local server but it's failing here
> > > 
> > > 
> > > Can you do a `yarn build` on your local machine?
> > 
> > 
> > I have done `yarn build`
> > image
> 
> Can you push again?

There nothing to push

**Comment by aaadddiii** [comment]
@HarshCasper  I can raise another pr after cloning the project once more

**Comment by HarshCasper** [comment]
> @HarshCasper I can raise another pr after cloning the project once more

Yeah. I'm closing this.

**Pull request closed without merging**